File: SpringBoot-SecKill-Scala/src/main/scala/cn/edu/jxnu/seckill/service/GoodsService.scala
package cn.edu.jxnu.seckill.service

import scala.language.implicitConversions
import scala.collection.JavaConversions._
import scala.collection.immutable.{ List => _, _ }
import java.util.{ List => JavaList }
import org.springframework.stereotype.Service
import org.springframework.beans.factory.annotation.Autowired
import cn.edu.jxnu.seckill.dao.GoodsDao
import cn.edu.jxnu.seckill.vo.GoodsVo
import cn.edu.jxnu.seckill.domain.SeckillGoods

/**
 * 商品服务层
 *
 * @author 梦境迷离.
 * @time 2018年5月19日
 * @version v1.0
 */
@Service
class GoodsService @Autowired() (goodsDao: GoodsDao) {

    /**
     * 查询全部
     */
    def listGoodsVo(): JavaList[GoodsVo] = goodsDao.listGoodsVo()

    /**
     * 根据商品id查询商品视图对象
     */
    def getGoodsVoByGoodsId(id: Long) = goodsDao.getGoodsVoByGoodsId(id)

    /**
     * 修改商品库存
     *
     * 默认每次减1
     */
    def reduceStock(goods: GoodsVo): Boolean = {
        var g = new SeckillGoods()
        g.setGoodsId(goods.getId())
        goodsDao.reduceStock(g) > 0
    }

    /**
     * 恢复库存
     */
    def resetStock(goodsList: JavaList[GoodsVo]) {
        for (goods <- goodsList) {
            val g = new SeckillGoods()
            g.setGoodsId(goods.getId())
            g.setStockCount(goods.getStockCount())
            goodsDao.resetStock(g)
        }
    }

}

File: seckill/src/main/java/cn/edu/jxnu/seckill/service/GoodsService.java
package cn.edu.jxnu.seckill.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import cn.edu.jxnu.seckill.dao.GoodsDao;
import cn.edu.jxnu.seckill.domain.SeckillGoods;
import cn.edu.jxnu.seckill.vo.GoodsVO;

@Service
public class GoodsService {
	@Autowired
	private GoodsDao goodsDao;

	public List<GoodsVO> listGoodsVO() {
		return goodsDao.listGoodsVO();
	}

	public GoodsVO getGoodsVOById(long goodsId) {
		return goodsDao.getGoodsVOByGoodsId(goodsId);
	}

	public boolean reduceStock(GoodsVO goods) {
		SeckillGoods g = new SeckillGoods();
		g.setGoodsId(goods.getId());
		int ret = goodsDao.reduceStock(g);
		return ret > 0;
	}

	public void resetStock(List<GoodsVO> goodsList) {
		for (GoodsVO goods : goodsList) {
			SeckillGoods g = new SeckillGoods();
			g.setGoodsId(goods.getId());
			g.setStockCount(goods.getStockCount());
			goodsDao.resetStock(g);
		}
	}
}
